src/function/table/range.cpp

```diff
@@ -1,269 +1,177 @@
 
 public:
-	bool Equals(const FunctionData &other_p) const override {
-		auto &other = (const RangeFunctionBindData &)other_p;
-		return other.start == start && other.end == end && other.increment == increment;
+	unique_ptr<FunctionData> Copy() const override {
+		return make_uniq<RangeFunctionBindData>(num_given_args, input_types, with_timestamps);
 	}
-};
 
-template <bool GENERATE_SERIES>
-static void GenerateRangeParameters(const vector<Value> &inputs, RangeFunctionBindData &result) {
-	for (auto &input : inputs) {
-		if (input.IsNull()) {
-			result.start = GENERATE_SERIES ? 1 : 0;
-			result.end = 0;
-			result.increment = 1;
-			return;
+	bool Equals(const FunctionData &other_p) const override {
+		auto &other = (const RangeFunctionBindData &)other_p;
+		if (with_timestamps != other.with_timestamps) {
+			return false;
 		}
+		if (num_given_args != other.num_given_args) {
+			return false;
+		}
+		if (input_types.size() != other.input_types.size()) {
+			return false;
+		}
+		for (idx_t i = 0; i < input_types.size(); i++) {
+			if (input_types[i] != other.input_types[i]) {
+				return false;
+			}
+		}
+		return true;
 	}
-	if (inputs.size() < 2) {
-		// single argument: only the end is specified
-		result.start = 0;
-		result.end = inputs[0].GetValue<int64_t>();
-	} else {
-		// two arguments: first two arguments are start and end
-		result.start = inputs[0].GetValue<int64_t>();
-		result.end = inputs[1].GetValue<int64_t>();
-	}
-	if (inputs.size() < 3) {
-		result.increment = 1;
-	} else {
-		result.increment = inputs[2].GetValue<int64_t>();
-	}
-	if (result.increment == 0) {
-		throw BinderException("interval cannot be 0!");
-	}
-	if (result.start > result.end && result.increment > 0) {
-		throw BinderException("start is bigger than end, but increment is positive: cannot generate infinite series");
-	} else if (result.start < result.end && result.increment < 0) {
-		throw BinderException("start is smaller than end, but increment is negative: cannot generate infinite series");
-	}
-}
+};
 
 template <bool GENERATE_SERIES>
 static unique_ptr<FunctionData> RangeFunctionBind(ClientContext &context, TableFunctionBindInput &input,
                                                   vector<LogicalType> &return_types, vector<string> &names) {
-	auto result = make_uniq<RangeFunctionBindData>();
-	auto &inputs = input.inputs;
-	GenerateRangeParameters<GENERATE_SERIES>(inputs, *result);
 
-	return_types.emplace_back(LogicalType::BIGINT);
+	LogicalType &first_arg_type = input.input_table_types[0];
+	idx_t num_given_args = input.input_table_types.size();
+
 	if (GENERATE_SERIES) {
-		// generate_series has inclusive bounds on the RHS
-		if (result->increment < 0) {
-			result->end = result->end - 1;
-		} else {
-			result->end = result->end + 1;
-		}
-		names.emplace_back("generate_series");
+		names.push_back("generate_series");
 	} else {
-		names.emplace_back("range");
+		names.push_back("range");
```

Comment:
Nit: Any reason this is changed from emplace to push?

Reply:
Didn't change that on purpose, I'll use `emplace`. :+1: